libs/sdk-ui-ext/src/internal/components/BaseVisualization.tsx

```diff
@@ -270,10 +257,40 @@ export class BaseVisualization extends React.PureComponent<IBaseVisualizationPro
         return !isEqual(omit(currentReferencePoint, "properties"), omit(nextReferencePoint, "properties"));
     }
 
+    private getVisualizationProps(): IVisProps {
+        return {
+            locale: this.props.locale,
+            dateFormat: this.props.dateFormat,
+            dimensions: {
+                width: this.props.width,
+                height: this.props.height,
+            },
+            custom: {
+                drillableItems: this.props.drillableItems,
+                totalsEditAllowed: this.props.totalsEditAllowed,
+            },
+            config: this.props.config,
+            theme: this.props.theme,
+            executionConfig: this.props.executionConfig,
+        };
+    }
+
     public getInsightWithDrillDownApplied(
         sourceVisualization: IInsight,
         drillDownContext: IDrillDownContext,
     ): IInsight {
         return this.visualization.getInsightWithDrillDownApplied(sourceVisualization, drillDownContext);
     }
+
+    public getExecution() {
+        if (!this.visualization) {
+            this.setupVisualization(this.props);
```

Comment:
Is this needed? The setupVisualization is called on mount, so when can this happen?

Reply:
Discussed offline, makes sense in case this is not used in React.

Reply:
Yes, in case it is used outside of react, you need to make sure, visualization is setup this way.